04-evm-mapping.md

```diff
@@ -0,0 +1,307 @@
+# FVM <> EVM mapping
+
+This document describes the mappings of structures, types, and procedures between the native FVM environment (WASM), and the EVM foreign VM (emulated).
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Integration](#integration)
+  - [State tree](#state-tree)
+  - [Mechanics and interfaces](#mechanics-and-interfaces)
+- [Contract memory model](#contract-memory-model)
+- [Account storage model](#account-storage-model)
+- [Addressing scheme](#addressing-scheme)
+  - [Proposed solution: universal stable addresses](#proposed-solution-universal-stable-addresses)
+- [Gas accounting and execution halt semantics](#gas-accounting-and-execution-halt-semantics)
+- [Ethereum logs/events](#ethereum-logsevents)
+- [Blockchain timing](#blockchain-timing)
+- [Precompiles](#precompiles)
+- [Cryptographic primitives](#cryptographic-primitives)
+- [Chain-specific opcodes](#chain-specific-opcodes)
+- [Cross-contract calls](#cross-contract-calls)
+- [References](#references)
+- [Annex A: Addressing solutions considered](#annex-a-addressing-solutions-considered)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Integration
+
+### State tree
+
+There won't be a segregated world state for Ethereum actors in the Filecoin blockchain. We will not use the Patricia Merkle Trie structure. Instead, EVM account state will be folded and docked into the Filecoin state tree.
+
+### Mechanics and interfaces
+
+EVM smart contracts (also known as EVM foreign actors in Filecoin terminology), are represented this way in the Filecoin state tree:
```

Comment:
Let's explicitly state that it's _just_ an actor.

Reply:
I.e., that the following object is a normal actor object (or just leave off the explicit actor struct).